mtp_api/apps/payment/views.py

```diff
@@ -1,14 +1,38 @@
 from django.http import Http404
-from rest_framework import mixins, viewsets
+import django_filters
+from rest_framework import mixins, viewsets, filters
 from rest_framework import status as http_status
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
 
-from mtp_auth.permissions import SendMoneyClientIDPermissions
+from core.permissions import ActionsBasedPermissions
+from mtp_auth.permissions import (
+    BankAdminClientIDPermissions, SendMoneyClientIDPermissions
+)
 from .exceptions import InvalidStateForUpdateException
-from .models import Payment
+from .models import Batch, Payment
 from .permissions import PaymentPermissions
-from .serializers import PaymentSerializer
+from .serializers import BatchSerializer, PaymentSerializer
+
+
+class BatchListFilter(django_filters.FilterSet):
+
+    class Meta:
+        model = Batch
+        fields = ('date',)
+
+
+class BatchView(
+    mixins.ListModelMixin, viewsets.GenericViewSet
+):
+    queryset = Batch.objects.all()
+    serializer_class = BatchSerializer
+    filter_backends = (filters.DjangoFilterBackend,)
+    filter_class = BatchListFilter
+
+    permission_classes = (
+        IsAuthenticated, ActionsBasedPermissions, BankAdminClientIDPermissions
```

Comment:
i think we need a subclass of `ActionsBasedPermissions` which includes 'view_batch' in the 'list' `actions_perms_map`